### DISCUSSION TITLE
VFD choices

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/core#270] [Q&A]
How does one determine if they have a HuanYang V1 or P2A VFD?  

I suspect I do not have V1 but my 2 VFDs are HY01D511B and HY02D223B.  Are these P2As? 

I am educating myself to write some user friendly VFD/Spindle documentation, by the way.

**Comment by terjeio** [comment]
Seems the best way to check is to test both and see which works (not raising alarm 14)?
The Huanyang documentation I have is useless, did you get any with your VFDs that documents the ModBus commands used?

**Reply by phil-barrett** [reply]
Pretty sure we have the same documentation!  Trial and error it is.  Of course, I'm in the process of debugging hardware at the same time but that is the story of my life...

**Comment by dresco** [comment]
I have two HY series VFDs here, both use the v1 protocol. 

My understanding of the various model lines is as follows;
- HY = v1 protocol
- P2 = v2 protocol
- GT = Modbus RTU compliant 
- FC = Modbus RTU compliant

**Reply by phil-barrett** [reply]
Thanks!  I'll try V1 then as both of mine are HY Series.
